### DISCUSSION TITLE
An Apple TV version will be excited.

### DISCUSSION THREAD

**Discussion by al03** [iina/iina#4094] [Ideas]
iina on Mac now is very powerful, compare to VLC, iina supported HDR and very friendly network video streaming, like m3u list, even HDR IPTV.
VLC has Apple TV version, iina not. And VLC on Apple TV is worse.
So I'm hoping iina can have an Apple TV version, so We can enjoy better video experience on TV.

**Comment by Zaxonov** [comment]
That would be lovely :D

**Comment by low-batt** [comment]
IINA is a full GUI for the [mpv](https://github.com/mpv-player/mpv) player specifically designed for macOS. IINA uses the macOS [AppKit](https://developer.apple.com/documentation/appkit) Framework. Apps for tvOS must use [UIKit](https://developer.apple.com/documentation/uikit). That would require so many changes that it is deemed better to implement that as a completely separate project.

**Reply by banannerrammers** [reply]
Is that something that is being worked on by anyone? And would it integrate/interact with IINA in any way, or would it just be a standalone app that wouldn't interact with IINA at all? Thanks!

**Reply by low-batt** [reply]
If someone were to write such an app it would be completely separate from IINA and would not integrate/interact with IINA.

For audio/video and subtitle rendering IINA is using a library from the [mpv](https://mpv.io/) project. There are other projects that use `libmpv` as well. The mpv project has a list of them [here](https://github.com/mpv-player/mpv/wiki/Applications-using-mpv#gui-frontends). One of them is [Outplayer](https://apps.apple.com/us/app/outplayer/id1449923287). I remember reading that the author was inspired by IINA. Outplayer is available for iPhone, iPad and Apple Vision, but not Apple TV. Maybe that will be added in the future. That is all I know about Outplayer. I've never tried it.

**Reply by fruwe** [reply]
Outplayer can stream to an Apple TV — and it does so impressively well. It’s the only option I’ve found that actually works

**Reply by low-batt** [reply]
Glad to hear from an actual Outplayer user that streaming to Apple TV works well.

There are requests for IINA to add direct support for [AirPlay](https://www.apple.com/airplay/), but at least when that was last investigated Apple was keeping details of the protocol private. So to get IINA to work with Apple TV you have to use the screen mirroring capabilites of macOS.

**Comment by NoX1De** [comment]
+1 PLEASE create a tvOS version!!!

**Comment by zee-pk** [comment]
I recently tried IINA, and I become a fan instantly, it is so good, specially when streaming from remote server (seek works like, reminds me of MPC-HC). I would love to have it on AppleTV.
